quality/integration/src/test/java/test/integration/blob/CasBlobStoreTest.java

```diff
@@ -249,36 +311,47 @@ private void verifyNotExists(String blobId) {
         } catch (BlobNotFoundException e) {
             // expected
         }
-        try {
-            _store.get(TABLE, blobId);
-            fail();
-        } catch (BlobNotFoundException e) {
-            // expected
-        }
     }
 
-    private void verifyPutAndGet(String blobId, final byte[] blobData, Map<String, String> attributes)
+    private void verifyPutAndGet(String blobId, final byte[] blobData, ImmutableMap<String, String> attributes)
             throws IOException {
-        _store.put(TABLE, blobId, new InputSupplier<InputStream>() {
-            @Override
-            public InputStream getInput() throws IOException {
-                return new ByteArrayInputStream(blobData);
-            }
-        }, attributes);
+        putBlob(blobId, blobData, attributes);
 
-        // verify that we can get what we put
         Blob blob = _store.get(TABLE, blobId);
+        // verify that we can get what we putBlob
+        verifyBlob(blob, blobId, blobData, attributes);
+
+        BlobMetadata blobMetadata = _store.getMetadata(TABLE, blobId);
+
+        //we don't check date, so can pass any non null value
+        BlobMetadata expectedBlobMetadata = getBlobMetadata(blobId, blobData, new Date(), attributes);
+        assertEqualsBlobMetadata(blobMetadata, expectedBlobMetadata);
+    }
+
+    private void verifyBlob(Blob blob, String blobId, byte[] blobData, Map<String, String> attributes) throws IOException {
         assertEquals(blob.getId(), blobId);
         assertEquals(blob.getLength(), blobData.length);
         assertEquals(blob.getAttributes(), attributes);
         ByteArrayOutputStream buf = new ByteArrayOutputStream();
         blob.writeTo(buf);
         assertEquals(buf.toByteArray(), blobData);
+    }
 
-        BlobMetadata md = _store.getMetadata(TABLE, blobId);
-        assertEquals(md.getId(), blobId);
-        assertEquals(md.getLength(), blobData.length);
-        assertEquals(md.getAttributes(), attributes);
+    private static void assertEqualsBlobMetadata(BlobMetadata actual, BlobMetadata expected) {
+        assertEquals(actual.getId(), expected.getId());
+        assertEquals(actual.getMD5(), expected.getMD5());
+        assertEquals(actual.getSHA1(), expected.getSHA1());
+        assertEquals(actual.getLength(), expected.getLength());
+        assertEquals(actual.getAttributes(), expected.getAttributes());
+    }
+
+    private void putBlob(String blobId, byte[] blobData, Map<String, String> attributes) throws IOException {
+        _store.put(TABLE, blobId, new InputSupplier<InputStream>() {
+            @Override
+            public InputStream getInput() throws IOException {
+                return new ByteArrayInputStream(blobData);
+            }
+        }, attributes);
```

Comment:
minor: for simplicity sake, use lambda: _store.put(TABLE, blobId, () -> new ByteArrayInputStream(blobData), attributes);

Reply:
was thinking that it's out of the scope of the ticket, but will fix in next commit